ml/src/main/scala/org/apache/spark/ml/FrameMLUtils.scala

```diff
@@ -0,0 +1,132 @@
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.spark.ml
+
+import org.apache.spark.h2o.H2OContext
+import org.apache.spark.ml.spark.models.MissingValuesHandling
+import org.apache.spark.mllib.linalg.Vectors
+import org.apache.spark.mllib.regression.LabeledPoint
+import org.apache.spark.rdd.RDD
+import org.apache.spark.sql.{Row, SQLContext}
+import org.apache.spark.sql.types.{DataTypes, StructField}
+import water.fvec.{Frame, H2OFrame}
+
+object FrameMLUtils {
+  /** 
+    * Converts a H2O Frame into an RDD[LabeledPoint]. Assumes that the last column is the response column which will be used 
+    * as the label. All other columns will be mapped to a vector and used as features. Categorical columns will be mapped to their 
+    * numerical values. 
+    * 
+    * @param frame Input frame to be converted 
+    * @param _response_column Column which contains the labels 
+    * @param nfeatures Number of features we want to use 
+    * @param missingHandler Missing values strategy 
+    * @param h2oContext Current H2OContext 
+    * @param sqlContext Current SQLContext 
+    * @return Returns an equivalent RDD[LabeledPoint] and means for each column 
+    */
+  def toLabeledPoints(frame: Frame,
+                      _response_column: String,
+                      nfeatures: Int,
+                      missingHandler: MissingValuesHandling,
+                      h2oContext: H2OContext,
+                      sqlContext: SQLContext): (RDD[LabeledPoint], Array[Double]) = {
+    var means: Array[Double] = new Array[Double](nfeatures)
+    val domains = frame.domains()
+
+    val trainingDF = h2oContext.asDataFrame(new H2OFrame(frame))(sqlContext)
+    val fields: Array[StructField] = trainingDF.schema.fields
+    var trainingRDD = trainingDF.rdd
+
+    if (MissingValuesHandling.Skip.eq(missingHandler)) {
+      trainingRDD = trainingRDD.filter(!_.anyNull)
+    } else if (MissingValuesHandling.MeanImputation.eq(missingHandler)) {
+      // Computing the means by hand and not using frame.means() as it does not compute the mean for enum columns
+      means = movingAverage(trainingRDD, fields, domains)
+    }
+
+    (trainingRDD.map(row => {
+      val features = new Array[Double](nfeatures)
```

Comment:
Instead of these 2 lines I like this more

   ``` val features = (0 until nfeatures).map{ i => if
        (row.isNullAt(i)) means(i) else toDouble(row.get(i), fields(i), domains(i))
      }.toArray[Double]``` It seems more functional to me since we set the features right away, but it's probably just matter of taste